rs/bitcoin/adapter/src/lib.rs

```diff
@@ -231,3 +236,47 @@ pub fn start_grpc_server_and_router(
         metrics_registry,
     );
 }
+
+/// The `main()` of the bitcoin adapter binary. This function is also called from
+/// the PocketIC server binary to run as a bitcoin adapter.
+pub async fn bitcoin_adapter_main() {
+    // We abort the whole program with a core dump if a single thread panics.
+    // This way we can capture all the context if a critical error
+    // happens.
+    abort_on_panic();
+
+    // We ignore arguments of the form `--run-as-bitcoin-adapter` passed by PocketIC.
+    let args = std::env::args().filter(|arg| arg != "--run-as-bitcoin-adapter");
+    let cli = Cli::parse_from(args);
```

Comment:
This part is changed in this PR, the rest is merely copied over from `/rs/bitcoin/adapter/src/main.rs`.